### PR TITLE
Add support for unfurling links

### PR DESCRIPTION
**Pull request by seanhoughton**
Hello, I have added support for the `chat.unfurl` method.

### CODE REVIEW

File: chat.go [hunk 1]
```diff
@@ -260,6 +266,19 @@ func MsgOptionDelete(timestamp string) MsgOption {
...
+// MsgOptionUnfurl unfurls a message based on the timestamp.
+func MsgOptionUnfurl(timestamp string, unfurls map[string]Attachment) MsgOption {
+	return func(config *sendConfig) error {
+		config.endpoint = APIURL + string(chatUnfurl)
+		config.values.Add("ts", timestamp)
+		unfurlsStr, err := json.Marshal(unfurls)
+		if err == nil {
+			config.values.Add("unfurls", string(unfurlsStr))
+		}
+		return nil
```
Comment (james-lawrence) [inline chat.go]: error should be returned in if occurs.

**Review by james-lawrence — CHANGES_REQUESTED** [review]
error handling needs fixing.

**Comment by seanhoughton** [comment]
Good catch, I've updated the code to return any error that might happen if the marshal fails.

**Review by james-lawrence — APPROVED** [review]

**Comment by james-lawrence** [comment]
thanks!

**Pull request merged**